src/biogeochem/CNVegNitrogenFluxType.F90

```diff
@@ -1034,6 +1039,50 @@ subroutine InitHistory(this, bounds)
        call hist_addfld1d (fname='NFERTILIZATION', units='gN/m^2/s', &
             avgflag='A', long_name='fertilizer added', &
             ptr_patch=this%fert_patch)
+       
+       this%repr_grainn_to_food_patch(begp:endp,:) = spval
+       this%repr_grainn_to_food_perharv_patch(begp:endp,:,:) = spval
+       this%repr_grainn_to_food_thisyr_patch(begp:endp,:) = spval
+       this%repr_grainn_to_seed_patch(begp:endp,:) = spval
+       do k = repr_grain_min, repr_grain_max
+          data1dptr => this%repr_grainn_to_food_patch(:,k)
+          call hist_addfld1d ( &
+               ! e.g., GRAINN_TO_FOOD
+               fname=get_repr_hist_fname(k)//'N_TO_FOOD', &
+               units='gN/m^2/s', &
+               avgflag='A', &
+               long_name=get_repr_longname(k)//' N to food', &
+               ptr_patch=data1dptr, &
+               default='inactive')
+          data1dptr => this%repr_grainn_to_seed_patch(:,k)
+          call hist_addfld1d ( &
+               ! e.g., GRAINN_TO_SEED
+               fname=get_repr_hist_fname(k)//'N_TO_SEED', &
+               units='gN/m^2/s', &
+               avgflag='A', &
+               long_name=get_repr_longname(k)//' N to seed', &
+               ptr_patch=data1dptr, &
+               default='inactive')
+          data2dptr => this%repr_grainn_to_food_perharv_patch(:,:,k)
+          call hist_addfld2d ( &
+               ! e.g., GRAINN_TO_FOOD_PERHARV
+               fname=get_repr_hist_fname(k)//'N_TO_FOOD_PERHARV', &
+               units='gN/m^2', &
+               type2d='mxharvests', &
+               avgflag='I', &
```

Comment:
I mentioned PR #2019 which, as written, eliminates 'I' hardwiring in the code; however, we are waiting for a different coding solution from CAM, so no need to worry about that right now.